File: src/main/java/com/example/reactcypressexperiment/BookStore.java
package com.example.reactcypressexperiment;

import org.springframework.stereotype.Repository;

import java.util.List;

import static java.util.Arrays.asList;

@Repository
public class BookStore {
    public List<Book> getAll() {
        return asList(get(1), get(2), get(3));
    }

    public Book get(int id) {
        switch (id) {
            case 1:
                return Book.builder().id(1).author("J.R.R. Tolkien").title("The Hobbit").build();
            case 2:
                return Book.builder().id(2).author("J.R.R. Tolkien").title("The Lord Of The Rings").build();
            case 3:
                return Book.builder().id(3).author("Steven King").title("It").build();
            default:
                throw new BookNotFoundException(id);
        }
    }
}

File: src/main/java/com/example/reactcypressexperiment/BookStore.kt
package com.example.reactcypressexperiment

import com.example.reactcypressexperiment.Book.Companion.NO_ID
import org.springframework.stereotype.Repository

@Repository
class BookStore {
    var all = mutableListOf<Book>()

    operator fun get(id: Int): Book = all.find { it.id == id } ?: throw BookNotFoundException(id)

    fun add(book: Book): Boolean =
        if (book.id == NO_ID) {
            book.id = maxId() + 1
            all.add(book)
            true
        } else {
            val i = all.indexOfFirst { it.id == book.id }
            if (i < 0) throw BookNotFoundException(book.id)
            all[i] = book
            false
        }

    private fun maxId(): Int = all.maxBy { it.id }?.id ?: 0
}
